go-controller/pkg/ovn/endpoints.go

```diff
@@ -52,60 +52,50 @@ func (ovn *Controller) AddEndpoints(ep *kapi.Endpoints) error {
 		// without a corresponding service.
 		klog.V(5).Infof("no service found for endpoint %s in namespace %s",
 			ep.Name, ep.Namespace)
-		return nil
+		return
 	}
 	if !util.IsClusterIPSet(svc) {
 		klog.V(5).Infof("Skipping service %s due to clusterIP = %q",
 			svc.Name, svc.Spec.ClusterIP)
-		return nil
+		return
 	}
 	klog.V(5).Infof("Matching service %s found for ep: %s, with cluster IP: %s", svc.Name, ep.Name,
 		svc.Spec.ClusterIP)
 
 	protoPortMap := ovn.getLbEndpoints(ep)
 	klog.V(5).Infof("Matching service %s ports: %v", svc.Name, svc.Spec.Ports)
-	for proto, portMap := range protoPortMap {
-		for svcPortName, lbEps := range portMap {
-			ips := lbEps.IPs
-			targetPort := lbEps.Port
-			for _, svcPort := range svc.Spec.Ports {
-				if svcPort.Protocol == proto && svcPort.Name == svcPortName {
-					if !ovn.SCTPSupport && proto == kapi.ProtocolSCTP {
-						klog.Errorf("Rejecting endpoint creation for unsupported SCTP protocol: %s, %s", ep.Namespace, ep.Name)
+	for _, svcPort := range svc.Spec.Ports {
+		if portMap, ok := protoPortMap[svcPort.Protocol]; ok {
+			if lbEps, ok := portMap[svcPort.Name]; ok {
+				if !ovn.SCTPSupport && svcPort.Protocol == kapi.ProtocolSCTP {
+					klog.Errorf("Rejecting endpoint creation for unsupported SCTP protocol: %s, %s", ep.Namespace, ep.Name)
+					continue
+				}
+				if util.ServiceTypeHasNodePort(svc) {
+					err = ovn.createGatewayVIPs(svcPort.Protocol, svcPort.NodePort, lbEps.IPs, lbEps.Port)
+					if err != nil {
+						klog.Errorf("Error in creating Node Port for svc %s, node port: %d - %v\n", svc.Name, svcPort.NodePort, err)
 						continue
 					}
-					if util.ServiceTypeHasNodePort(svc) {
-						klog.V(5).Infof("Creating Gateways IP for NodePort: %d, %v", svcPort.NodePort, ips)
```

Comment:
why remove the debug log?

Reply:
Because `createGatewayVIPs` logs the same message on its first line.

Reply:
ok cool